toolkit/tools/internal/rpm/rpm_test.go

```diff
@@ -151,3 +151,45 @@ func TestShouldNotFindCheckSectionInSpecWithoutCheckSection(t *testing.T) {
 	assert.NoError(t, err)
 	assert.False(t, hasCheckSection)
 }
+
+func TestExtractNameFromRPMPath(t *testing.T) {
+	tests := []struct {
+		name     string
+		rpmFile  string
+		expected string
+		err      error
+	}{
+		{
+			name:     "valid rpm file",
+			rpmFile:  "/path/to/pkg-1.0.0-1.noarch.rpm",
+			expected: "pkg",
+			err:      nil,
+		},
+		{
+			name:     "valid rpm file with complex name",
+			rpmFile:  "/path/to/pkg-name-1.0.0-1.noarch.rpm",
+			expected: "pkg-name",
+			err:      nil,
+		},
+		{
+			name:     "invalid rpm file",
+			rpmFile:  "/path/to/garbage.rpm",
+			expected: "",
+			err:      fmt.Errorf("invalid RPM file name '%s', can't extract name", "/path/to/garbage.rpm"),
+		},
+		{
+			name:     "empty rpm file",
+			rpmFile:  "",
+			expected: "",
+			err:      fmt.Errorf("invalid RPM file name '%s', can't extract name", ""),
+		},
+	}
```

Comment:
Could we instead split these into separate test cases? This way, it's going to be easier to tell, what went wrong, if one of them fails, since the failing case will be encoded in the name. Also, this will make the tests slightly easier to analyze, although I'd agree this one is pretty clear already.

Reply:
Oh, neat, I didn't know that.